homeassistant/components/hue_ble/config_flow.py

```diff
@@ -0,0 +1,151 @@
+"""Config flow for Hue BLE integration."""
+
+from __future__ import annotations
+
+from enum import Enum
+import logging
+from typing import Any
+
+from HueBLE import HueBleLight
+import voluptuous as vol
+
+from homeassistant.components import bluetooth
+from homeassistant.components.bluetooth.api import (
+    async_ble_device_from_address,
+    async_scanner_count,
+)
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_MAC, CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+
+from .const import DOMAIN, URL_PAIRING_MODE
+from .light import get_avaliable_color_modes
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def validate_input(hass: HomeAssistant, address: str) -> Error | None:
+    """Return error if cannot connect and validate."""
+
+    ble_device = async_ble_device_from_address(hass, address.upper(), connectable=True)
+
+    if ble_device is None:
+        count_scanners = async_scanner_count(hass, connectable=True)
+        _LOGGER.debug("Count of BLE scanners in HA bt: %i", count_scanners)
+
+        if count_scanners < 1:
+            return Error.NO_SCANNERS
+        return Error.NOT_FOUND
+
+    try:
+        light = HueBleLight(ble_device)
+
+        await light.connect()
+
+        if light.authenticated is None:
+            _LOGGER.warning(
+                "Unable to determine if light authenticated, proceeding anyway"
+            )
+        elif not light.authenticated:
+            return Error.INVALID_AUTH
+
+        if not light.connected:
+            return Error.CANNOT_CONNECT
+
+        if len(get_avaliable_color_modes(light)) == 0:
+            return Error.NOT_SUPPORTED
+
+        _, errors = await light.poll_state()
+        if len(errors) != 0:
+            _LOGGER.warning("Errors raised when connecting to light: %s", errors)
+            return Error.CANNOT_CONNECT
+        else:  # noqa: RET505
+            return None
```

Comment:
Why do you ignore the lint error?
I'd suggest to adjust the code instead:
```suggestion
        return None
```